### PR TITLE
virttest.utils_net: support blocking specific host with certain time

### PR DESCRIPTION
**Pull request by chuanchang**
As subject.
Signed-off-by: Alex Jia <ajia@redhat.com>

### CODE REVIEW

File: virttest/utils_net.py [hunk 1]
```diff
@@ -3014,3 +3014,33 @@ def check_listening_port_remote_by_service(server_ip, server_user, server_pwd,
     except:
         if session:
             session.close()
+
+
+def block_specific_ip_by_time(ip_addr, block_time=None, runner=None):
```
Comment (will-Do) [inline virttest/utils_net.py]: Good func, just a tiny suggestion about the func name, as the ```block_time``` could be omitted by caller and the method you use is by iptables, so my suggestion is ```block_ip_by_iptables```, what do you think?

Comment (chuanchang) [inline virttest/utils_net.py]: @will-Do, the name 'block_ip_by_iptables' is too common, which isn't what I want, this funcs is used for blocking specific host IP with certain time.

**Comment by lmr** [comment]
You have to fix your docstrings. Per Travis:

        2 ERRORs/WARNINGs detected while building the html docs:
        1) /home/travis/build/autotest/virt-test/virttest/utils_net.py:docstring of virttest.utils_net.block_specific_ip_by_time:4: ERROR: Unexpected indentation.
        2) /home/travis/build/autotest/virt-test/virttest/utils_net.py:docstring of virttest.utils_net.block_specific_ip_by_time:6: WARNING: Block quote ends without a blank line; unexpected unindent.

**Comment by will-Do** [comment]
@waynesun09 @Hao-Liu 
As ajia is need this PR merged for the jenkins job, so please help to reveiw it

File: virttest/utils_net.py [hunk 2]
```diff
@@ -3028,3 +3028,28 @@ def check_listening_port_remote_by_service(server_ip, server_user, server_pwd,
...
+    N >=1, the ${time} is [hours|minutes|seconds],etc, the default is '1 seconds'
+    :param runner: command runner, it's a remote session
+    """
+    cmd = "iptables -A INPUT -s %s -m time --kerneltz --timestart \
+           $(date +%%H:%%M:%%S) --timestop $(date --date='+%s' +%%H:%%M:%%S) \
+           -j DROP" % (ip_addr, block_time)
+    list_rules = "iptables -L"
+    try:
+        if runner is None:
+            runner = local_runner
```
Comment (PandaWei) [inline virttest/utils_net.py]: It's more reasonable that move the judgment out of try/catch.

Comment (chuanchang) [inline virttest/utils_net.py]: @PandaWei，older python cannot support the full exception syntax. please read 'Catching Exceptions' section in CODING_STYLE under the virt-test directory.

 The accepted formats for exception catching are:
     try...except...
     try...finally...
     try...except...else...
     try...finally...else...

Comment (PandaWei) [inline virttest/utils_net.py]: Maybe There's some mistake
I mean that you *don't* need to check the runner is None or not  *in the try/catch block.

Comment (chuanchang) [inline virttest/utils_net.py]: @PandaWei, it's a condition to check local 'iptables' command.

Comment (PandaWei) [inline virttest/utils_net.py]: Nope....
You don't understand what i want to say...
I mean the check for runner would never throw any except.
you should check it before try/catch block.
Please refer to virttest/utils_net.py for more cases.

File: virttest/utils_net.py [hunk 3]
```diff
@@ -3028,3 +3028,28 @@ def check_listening_port_remote_by_service(server_ip, server_user, server_pwd,
...
+            try:
+                os_dep.command("iptables")
+            except ValueError, details:
+                raise error.TestNAError(details)
```
Comment (PandaWei) [inline virttest/utils_net.py]: and  I think it's better to check iptables exists or not at the beginning of block_specific_ip_by_time.
all of steps that before os_dep.command("iptables"),  turn meaningless if iptables doesn't exist.

**Comment by lmr** [comment]
Hello @chuanchang, if you could be so kind, please rebase it and re-open it against

https://github.com/avocado-framework/avocado-vt

Thanks!

**Pull request closed without merging**